lib/services/nginx.pm

```diff
@@ -0,0 +1,94 @@
+# SUSE's openQA tests
+#
+# Copyright 2019-2020 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: nginx
+# Summary: Package for nginx service tests
+#
+# Maintainer: Pavel Dostal <pdostal@suse.cz>
+
+package services::nginx;
+use base 'opensusebasetest';
+use testapi;
+use warnings;
+use strict;
+use utils qw(zypper_call common_service_action script_retry);
+
+my $service_type = 'Systemd';
+
+sub install_service {
+    zypper_call 'in nginx';
+}
+
+sub enable_service {
+    common_service_action 'nginx', $service_type, 'enable';
+}
+
+sub start_service {
+    common_service_action 'nginx', $service_type, 'start';
+}
+
+# Configure nginx so it can be tested
+sub config_service {
+    zypper_call('in curl') if (script_run('which curl') != 0);
+    zypper_call('in openssl') if (script_run('which openssl') != 0);
+
+    assert_script_run('echo "<html>Hello from nginx</html>" > /srv/www/htdocs/index.html');
+
+    my $openssl_command = 'openssl req -keyout /etc/nginx/ssl/key.pem -out /etc/nginx/ssl/cert.pem';
+    $openssl_command .= ' -x509 -newkey rsa:4096 -sha256 -days 365 -nodes -subj "/CN=vhost"';
+    assert_script_run('mkdir -p /etc/nginx/ssl/');
+    assert_script_run($openssl_command);
+
+    # Add new virtual host and check the configuration files
+    assert_script_run('curl -fv ' . data_url('console/nginx_vhost.conf') . ' -o /etc/nginx/vhosts.d/nginx_vhost.conf');
+    assert_script_run('nginx -t');
+
+    assert_script_run "echo '127.0.0.1 vhost' >> /etc/hosts";
+    assert_script_run "echo '::1 vhost' >> /etc/hosts";
+
+    common_service_action 'nginx', $service_type, 'reload';
+}
+
+# check service is running and enabled
+sub check_service {
+    common_service_action 'nginx', $service_type, 'is-enabled';
+    common_service_action 'nginx', $service_type, 'is-active';
+}
+
+sub check_function {
+    my $grep = ' | grep "Hello from nginx"';
+    # nginx does not respond right away after service restart
+    #script_retry('curl http://localhost/' . $grep, delay => 3, retry => 3);
```

Comment:
Don't add disabled code as nobody but you will understand why this is not active. Better leave it out and keep in your local git repository, either in `git stash` or a temporary "WIP"-commit

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details